### ISSUE TITLE
Can I add variables through template literal?

### ISSUE THREAD

**Issue by kristojorg** [apollographql/graphql-tag#73]
I am trying to write a complex mutation with pieces that are dynamically created. I'm wondering: 

(a) Is it okay to do something like this with the dynamic creation of mutations,

(b) can I use variables from js directly in my template literals? Or do I need to pass them as graphQL variables? ex:

```js
const createEvent = (existing, event) => gql`
  mutation logEvent {
    createEvent(
      channel: "${event.channel}"
      metadata: ${JSON.stringify(event.metadata)}
      segmentId: "${event.messageId}"
      timestamp: "${event.timestamp}"
      type: ${event.type === 'page' ? 'PAGEVIEW' : 'TRACK'}
      userId: "${event.anonymousId}"
      ${createProjectVersion(existing, event.properties.version)}
      ${createPage(existing, event.context.page)}
      name: "${event.name}"
      ${createIpAddress(existing, event)}
    ) {
      id
    }
  }
`;

const createPage = (existing, page) => {
  if (existing.pages[0]) return `pageId: "${existing.pages[0].id}"`;
  return `
    page: {
      path: "${page.path}",
      url: "${page.url}",
      title: "${page.title}"
    }
  `;
};
```

here you can see that `createPage` is dynamically creating either a nested update or a reference to a page. I am seeing an error on `metadata` that it expected type `Json!` but got `undefined`. I'm guessing this is because it is not a graphQL defined variable? Or maybe my JSON isn't parsing correctly. Any help here would be appreciated.

**Comment by jnwng** [comment]
apologies for the brevity but just writing this while in-transit. a couple of notes.

let me preface this by stating that yes, you may interpolate any sort of values into your `gql` document definition, its just a string after all. 
however, _should_ you interpolate values? i'd probably say, no, you probably don't need to. straight from the graphql specification, [variables](https://facebook.github.io/graphql/#sec-Language.Variables) are probably what you want:
> A GraphQL query can be parameterized with variables, maximizing query reuse, and avoiding costly string building in clients at runtime.

what are some other benefits of having a static query? the query that you provided as an example cannot be statically analyzed to make sure that it conforms to the schema that exists on your backend. tools like [eslint-plugin-graphql](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0ahUKEwj6kc-66pLTAhUrilQKHXqfDBcQFggcMAA&url=https%3A%2F%2Fgithub.com%2Fapollographql%2Feslint-plugin-graphql&usg=AFQjCNGRPu768Fy1zIoaYUJlUn_eCmtIwQ&sig2=GGLzoe8e3nyEug5Wv-_3kA) will throw errors on this, because those linters don't know what the runtime values of those interpolations will be. see more detail on this in this [wonderful article](https://dev-blog.apollodata.com/5-benefits-of-static-graphql-queries-b7fa90b0b69a) on the benefits of static queries. and on the plus-side, once your query is static, you can just paste this into graphiql to validate that it works with your backend (instead of having to always test it in your application)

you can also use [input objects](https://facebook.github.io/graphql/#sec-Input-Object-Values) to pass your mutation inputs as objects so you dont have to be super verbose about your mutation arguments.

in summary, yes, you could interpolate values (if you wanted), but you lose a lot of the benefits of graphql that i think are quite valuable.

**Comment by jnwng** [comment]
closing this issue as i dont think there's a particular issue with `graphql-tag` here, you may want to double check that all of your interpolated values are returning valid strings.

**Comment by svendpeter** [comment]
Hello ,
Can you kindly help me with a problem , maybe I am using the graphql-tag in a wrong way ...
I am trying to understand something , and this seems to me the best place to get an answer to the question .
I am building an app that loads graphQl queries from database and then tries to execute them .
The queries are being red from database as strings, as follows : gql`query nameOfTheQuery.....` .My problem is that this queries can not be executed , I think because they are strings and not query objects . The string is not parsed to a query . 

Would it help if I just load the string from the database in this form  "query naamOfTheQuery ...." and then interpolate it , on the front end , between gql`  ` ?Would this parse the string into a query object ?
 I've spent a few hours on this bug , and this seems to be the solution .
Thank you ..
